ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryEngine.java

```diff
@@ -56,12 +56,13 @@ static OutputNode buildQueryLogicalPlan(
       final Query query,
       final Optional<Sink> sink,
       final MetaStore metaStore,
-      final KsqlConfig config
+      final KsqlConfig config,
+      final SessionConfig sessionConfig
   ) {
     final String outputPrefix = config.getString(KsqlConfig.KSQL_OUTPUT_TOPIC_NAME_PREFIX_CONFIG);
 
     final QueryAnalyzer queryAnalyzer =
-        new QueryAnalyzer(metaStore, outputPrefix, config);
+        new QueryAnalyzer(metaStore, outputPrefix, config.cloneWithPropertyOverwrite(sessionConfig.getOverrides()));
```

Comment:
Is there a reason we can't just use `sessionConfig.getConfig(true)` directly?

Also, if this is the only thing we need to pipe the session config through for, we could also just pass `KsqlConfig configWithOverrides` directly.

Reply:
I think both of these points are good, pushing changes with the fixes now.